### ISSUE TITLE
Exclude archived content from default ActiveQueryContent queries

### ISSUE THREAD

**Issue by buddh4** [humhub/humhub#3947]
Queries like `Calendar::find()->readable()` by default include archived entries and there is no build-in way of excluding archived entries. Usually you don't want to include archived entries and either only display archived entries or explicitly include archived entries e.g. by filter.

Perhaps a query like the following to include archived entries would solve the issue:

```php
Calendar::find()->readable()->archived()
```

**Comment by buddh4** [comment]
Example:

```
 $query = Task::find()->readable()
            ->andWhere('content.archived = 0')
            ->andWhere('space.status = 1 OR space.status IS NULL');
```

**Comment by luke-** [comment]
Hmm, what about other similar future cases:
- Deleted State (To provide something like a trash can)
- Draft State (or Publish Time)
- Content of blocked/disabled/ignored users
- ?

Instead adding this to `readable()` we could add a new state  "visible()"?
This could then include `readable()` and also the transition should also be very easy.

**Comment by buddh4** [comment]
Yes, we should add a seperate helper function for this. One issue I see that we have to keep track of the joins, but this should not be a big problem.

Also in case we filter for a specific space, and this space is archived, we should not check for the space status as in my example above. I like the `visible()` idea we could add both `visible()` and `archived()` for more fine grained control. Somethimes you may want to filter for only archived content.
